content/en/docs/tasks/administer-cluster/encrypt-data.md

```diff
@@ -261,66 +421,139 @@ program to retrieve the contents of your secret data.
+
+## Rotate a decryption key {#rotating-a-decryption-key}
+
+Changing an encryption key for Kubernetes without incurring downtime requires a multi-step operation,
+especially in the presence of a highly-available deployment where multiple `kube-apiserver` processes
+are running.
+
+1. Generate a new key and add it as the second key entry for the current provider on all
+   control plane nodes.
+1. Restart **all** `kube-apiserver` processes, to ensure each server can decrypt
+   any data that are encrypted with the new key.
+1. Make a secure backup of the new encryption key. If you lose all copies of this key you would
+   need to delete all the resources were encrypted under the lost key, and workloads may not
+   operate as expected during the time that at-rest encryption is broken.
+1. Make the new key the first entry in the `keys` array so that it is used for encryption-at-rest
+   for new writes
+1. Restart all `kube-apiserver` processes to ensure each control plane host now encrypts using the new key
+1. As a privileged user, run `kubectl get secrets --all-namespaces -o json | kubectl replace -f -`
+   to encrypt all existing Secrets with the new key
+1. After you have updated all existing Secrets to use the new key and have made a secure backup of the
+   new key, remove the old decryption key from the configuration.
+
+## Decrypt all data {#decrypting-all-data}
+
+This example shows how to stop encrypting the Secret API at rest. If you are encrypting
+other API kinds, adjust the steps to match.
+
+To disable encryption at rest, place the `identity` provider as the first
+entry your encryption configuration file:
+
+```yaml
+---
+apiVersion: apiserver.config.k8s.io/v1
+kind: EncryptionConfiguration
+resources:
+  - resources:
+      - secrets
+      # list any other resources here that you previously were
+      # encrypting at rest
+    providers:
+      - identity: {} # add this line
+      - aescbc:
+          keys:
+            - name: key1
+              secret: <BASE 64 ENCODED SECRET> # keep this in place
+                                               # make sure it comes after "identity"
+```
+
+Then run the following command to force decryption of all Secrets:
 
 ```shell
 kubectl get secrets --all-namespaces -o json | kubectl replace -f -
 ```
 
+Once you have replaced all existing encrypted resources with backing data that
+don't use encryption, you can remove the encryption settings from the
+`kube-apiserver`.
+
+## Configure automatic reloading
+
+You can specify a boolean `--encryption-provider-config-automatic-reload` command line
+argument to `kube-apiserver`. That setting determines whether the API server should automatically
+reload the file you specify for `--encryption-provider-config`. Enabling this option allows you
+to change the keys for encryption at rest without restarting the API server.
+
+To allow automatic reloading, configure the API server to run with:
+`--encryption-provider-config-automatic-reload=true`
```

Comment:
This is again redundant.

Reply:
I disagree, because we haven't told readers that it's an argument that takes a boolean value. The reader can infer it, but in tech docs it pays to be precise.


OK, OK, it doesn't pay very well if your an open source contributor - but I hope you get my point.